modules/data.land/R/soilgrids_soc_extraction.R

```diff
@@ -184,11 +226,17 @@ soilgrids_soilC_extract <- function (site_info, outdir=NULL, verbose=TRUE) {
   rownames(soilgrids_soilC_data) <- NULL
 
   if (!is.null(outdir)) {
-    PEcAn.logger::logger.info(paste0("Storing results in: ",file.path(outdir,"soilgrids_soilC_data.csv")))
-    utils::write.csv(soilgrids_soilC_data,file=file.path(outdir,"soilgrids_soilC_data.csv"),row.names = FALSE)
-  }
-  else {
-    PEcAn.logger::logger.error("No output directory found.")
+    # Ensure the directory exists; create if not
+    if (!dir.exists(outdir)) {
+      dir.create(outdir, recursive = TRUE)
+      PEcAn.logger::logger.info(paste0("Created output directory: ", outdir))
+    }
+    PEcAn.logger::logger.info(paste0("Storing results in: ", file.path(outdir, "soilgrids_soilC_data.csv")))
```

Comment:
- Logger functions do the pasting for you
- No need to separately call out creating the directory -- reporting it in the result path is enough
```suggestion
      dir.create(outdir, recursive = TRUE)
    }
    PEcAn.logger::logger.info("Storing results in: ", file.path(outdir, "soilgrids_soilC_data.csv"))
```